### PR TITLE
31 make mobile home page

### LINKED ISSUE #31: Make Mobile Home page

### PR DESCRIPTION
**Pull request by Gormiee**
Added scuffed homepage, No more stupidstore => JsonPublisher, using core base components in the mobile app

### CODE REVIEW

File: src/UICrafter.Core/Components/BaseComponents/BaseLoadMetaDataList.razor [hunk 1]
```diff
@@ -37,6 +37,27 @@ else
     protected virtual void HandleClick(AppView view) { }
     protected virtual string? SetupHref(AppView view) { return null; }
 
+    // protected override async void OnParametersSet()
```
Comment (Nephim) [inline src/UICrafter.Core/Components/BaseComponents/BaseLoadMetaDataList.razor]: ??

File: src/UICrafter.Mobile/Components/Pages/Home.razor [hunk 2]
```diff
@@ -1,5 +1,52 @@
...
+                </MudButton>
+            </MudStack>
+            </NotAuthorized>
+
+    </AuthorizeView>  
+</MudContainer>
+
+@code {
+ 
+    private void NavigateToLogin()
```
Comment (Nephim) [inline src/UICrafter.Mobile/Components/Pages/Home.razor]: We have talked about this! HREF!

File: src/UICrafter.Mobile/Components/UIComponents/MobileAPIButton.razor [hunk 3]
```diff
@@ -1,57 +1,58 @@
...
-            buttonColor = Color.Primary;
+            //buttonColor = Color.Primary;
             _processing = true;
             var originalLabel = Button.Button.Label;
 
-            // Assume that Button.URL is the API endpoint
-            var response = await httpclient.GetStringAsync(Button.Button.URL);
+            var response = await httpClient.GetStringAsync(Button.Button.URL);
 
             if (response == null)
```
Comment (Nephim) [inline src/UICrafter.Mobile/Components/UIComponents/MobileAPIButton.razor]: I dont think that this is correct. I am not aware of this ever returning a null object. If you want to check if a response went well there are IsSuccess and there is ThrowIfError. Depending on how you want to check what went wrong.

File: src/UICrafter.Mobile/Components/_Imports.razor [hunk 4]
```diff
@@ -18,3 +18,4 @@
 @using Serilog
 @using Google.Protobuf
 @using System.Security.Claims
+@using UICrafter.Core.Components.BaseComponents
```
Comment (Nephim) [inline src/UICrafter.Mobile/Components/_Imports.razor:21]: Good job

File: src/UICrafter.Core/Components/BaseComponents/BaseAPIButton.razor [hunk 5]
```diff
@@ -1,11 +1,12 @@
 <MudContainer Class="d-flex justify-center pa-1">
-    <MudButton Class="mb-2" Variant="Variant.Filled" OnClick="OnClick" @onfocus="OnFocus">
+    <MudButton Class="mb-2" Variant="Variant.Filled" Color="@buttonColor" OnClick="OnClick" @onfocus="OnFocus">
         @ButtonText
     </MudButton>
 </MudContainer>
 
 @code {
     protected string ButtonText { get; set; } = string.Empty;
+    protected Color buttonColor;
```
Comment (Nephim) [inline src/UICrafter.Core/Components/BaseComponents/BaseAPIButton.razor]: Naming

File: src/UICrafter.Mobile/Components/UIComponents/MobileTextBox.razor [hunk 6]
```diff
@@ -1,20 +1,28 @@
-<MudTextField @bind-Value="Textbox.Textbox.Content"
-              Label="@Textbox.Textbox.Label"
-              Variant="Variant.Outlined"
-              ReadOnly="true"
-              MaxLines="@Textbox.Textbox.NumberOfLines"
-              AutoGrow="true">
-</MudTextField>
+@using System.Text.Json
+@using UICrafter.Core.Components.BaseComponents
```
Comment (Nephim) [inline src/UICrafter.Mobile/Components/UIComponents/MobileTextBox.razor]: Are all of these usings needed? You just added one of them atleast?

File: src/UICrafter.Mobile/Components/Pages/Home.razor [hunk 7]
```diff
@@ -1,5 +1,52 @@
 @page "/"
+@inject NavigationManager Navigation
```
Comment (Nephim) [inline src/UICrafter.Mobile/Components/Pages/Home.razor]: You dont need these injects

File: src/UICrafter.Mobile/Components/Pages/AppViewPage.razor [hunk 8]
```diff
@@ -1,4 +1,5 @@
 @page "/appview/{viewId:long?}"
+@attribute [Authorize]
 
 @using UICrafter.Mobile.Components.UIComponents
```
Comment (Nephim) [inline src/UICrafter.Mobile/Components/Pages/AppViewPage.razor]: Imports?

**Review by Nephim — CHANGES_REQUESTED** [review]
Plz fix

File: src/UICrafter.Core/Components/BaseComponents/BaseLoadMetaDataList.razor (hunk 1 above)
Comment (Aagaard00) [reply src/UICrafter.Core/Components/BaseComponents/BaseLoadMetaDataList.razor]: Agree

**Review by Aagaard00 — APPROVED** [review]
LGTM, endelig væk med POC og Stupid :))

**Pull request merged**